### DISCUSSION TITLE
Why doesn't this library set the "alg" header when generating a key pair?

### DISCUSSION THREAD

**Discussion by treeder** [panva/jose#648] [Q&A]
You can't import a key without knowing the alg, if you try you get a:

```
Error: "alg" argument is required when "jwk.alg" is not present
```

Example:

```
    let key = await jose.generateKeyPair('EdDSA', { extractable: true })
    let privateKey = await jose.exportJWK(key.privateKey)
    console.log("importing right after exporting...")
    let ikey = await jose.importJWK(privateKey) // errors here
    console.log("imported key", ikey)
```

Is there a reason not to set it?

**Comment by panva** [comment]
> Is there a reason not to set it?

Because it couldn't be made consistent between the different algorithms and runtimes. ECDH keys don't map to a single algorithm in WebCryptoAPI, node:crypto KeyObject has no notion of an algorithm.

**Reply by treeder** [reply]
Ok, makes sense. But what about for the algorithms that do map properly? Seems like it could set it for those without issues?

**Reply by panva** [reply]
That would really just be `EdDSA` because for instance 'rsa' KeyObject instance is applicable to any RS* and PS* algorithm. EdDSA also will be deprecated soon in favour of Ed25519/Ed448 algorithm identifier respectively.
